lib/checkCriticalPullRequest.js

```diff
@@ -0,0 +1,137 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview File to handle checks when a PR adds/modifies a model.
+ */
+
+const {
+  CRITICAL_LABEL,
+  getAllChangedFiles,
+  getNameString,
+  getNewItemsFromFileByRegex,
+  hasCriticalLabel
+} = require('./utils');
+
+const { teamLeads } = require('../userWhitelist.json');
+const STORAGE_DIR_PREFIX = 'core/storage/';
+const MODEL_REGEX = /(?<addition>\+)(?<classDefinition>class\s)(?<name>[a-zA-Z]{2,256})(?<suffix>Model)(?<funDef>\()/;
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem} file
+ */
+const getNewModelsFromFile = (file) => {
+  return getNewItemsFromFileByRegex(MODEL_REGEX, file);
+};
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem} file
+ */
+const addsNewModel = (file) => {
+  const newModels = getNewModelsFromFile(file);
+  return newModels.length > 0;
+};
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem[]} newModelFiles
+ */
+const getModelNameString = (newModelFiles) => {
+  return getNameString(
+    newModelFiles,
+    {
+      singular: 'model',
+      plural: 'models'
+    },
+     MODEL_REGEX
+  );
+};
+
+/**
+ * @param {string} filename
+ */
+const isInStorageDir = (filename) => {
+  return filename.startsWith(STORAGE_DIR_PREFIX);
+};
+
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem[]} newModelFiles
+ * @param {import('probot').Octokit.PullsListFilesResponseItem[]} changedFiles
+ * @param {string} prAuthor - Author of the Pull Request
+ */
+const getCommentBody = (newModelFiles) => {
```

Comment:
Let us add this to function to utils as well, you can pass the name of the group/person to ping.